app/src/test/kotlin/com/thundermaps/apilib/android/impl/resources/ChannelResourceImplTest.kt

```diff
@@ -0,0 +1,290 @@
+
+        val result =
+            channelResource.getChannels(defaultParameters, teamId = TEAM_ID)
+
+        verifyAndroidClient(4)
+        assertTrue(result.isError)
+        val error =
+            ((result as Result.Error).exception as ResponseException).responseError
+        assertEquals(
+            "bad_credentials",
+            error.errorCodes?.base?.firstOrNull()?.error
+        )
+        assertTrue(inspectCalled)
+    }
+
+    private fun verifyAndroidClient(expectedVersion: Int) {
+        val parameterCaptor = argumentCaptor<RequestParameters>()
+        verify(androidClient).client(parameterCaptor.capture())
+        val requestParameters = parameterCaptor.firstValue
+        assertEquals(
+            expectedVersion,
+            requestParameters.api_version
+        )
+        assertEquals(
+            saferMeCredentials,
+            requestParameters.credentials
+        )
+    }
+
+    companion object {
+        private const val TEAM_ID = 6688L
+        private val responseHeaders =
+            headersOf("Content-Type" to listOf(ContentType.Application.Json.toString()))
+        private const val APPLICATION_ID = "com.thundermaps.saferme"
+        private const val CHANNELS_PATH =
+            "/api/v4/teams/$TEAM_ID/channels?fields=${ChannelResource.DEFAULT_FIELDS}"
+        private const val TEST_KEY = "Test Key"
+        private const val TEST_INSTALL = "Install App"
+        private const val TEST_APP = APPLICATION_ID
+        private const val TEST_TEAM = "Test Team"
+        private const val TEST_CLIENT_UUID = "client uuid"
+        private val saferMeCredentials =
+            SaferMeCredentials(TEST_KEY, TEST_INSTALL, TEST_APP, TEST_TEAM, TEST_CLIENT_UUID)
+        private val defaultParameters = RequestParameters(
+            customRequestHeaders = HashMap(),
+            credentials = saferMeCredentials,
+            host = Staging.servers.first(),
+            port = null,
+            api_version = 4
+        )
+
+        private val channel = Channel(
+            id = 30108L,
+            allowPublicViewers = false,
+            allowPublicComments = true,
+            allowUserDeleteOwnReports = false,
+            areNewReportsAnonymous = true,
+            isBannersEnabled = false,
+            categoryId = 193563L,
+            description = "",
+            isFormLocked = false,
+            isHazardChannel = false,
+            isAddonChannel = false,
+            isDeletableBy = false,
+            isManageableBy = true,
+            isOperableBy = true,
+            isReportableBy = true,
+            lastReportDate = "2021-10-11T12:57:42+13:00",
+            logo = Avatar(
+                mini = "missing/logos/mini.png",
+                small = "missing/logos/small.png",
+                medium = "missing/logos/medium.png",
+                large = "missing/logos/large.png",
+                huge = "missing/logos/huge.png"
+            ),
```

Comment:
I believe channels logos are being removed and we won't need to store them in the app